src/ui/connection.ts

```diff
@@ -0,0 +1,200 @@
+import { commands, env, Uri, window } from "vscode";
+import IBMi, { ConnectionErrorCode, ConnectionMessageType } from "../api/IBMi";
+
+export function messageCallback(type: ConnectionMessageType, message: string) {
+  switch (type) {
+    case `info`:
+      window.showInformationMessage(message);
+      break;
+    case `warning`:
+      window.showWarningMessage(message);
+      break;
+    case `error`:
+      window.showErrorMessage(message);
+      break;
+  }
+}
+
+export async function handleConnectionResults(connection: IBMi, error: ConnectionErrorCode, data: any) {
+  switch (error as ConnectionErrorCode) {
+    case `shell_config`:
+      const chosen = await window.showInformationMessage(`Error in shell configuration!`, {
+        detail: [
+          `This extension can not work with the shell configured on ${connection.currentConnectionName},`,
+          `since the output from shell commands have additional content.`,
+          `This can be caused by running commands like "echo" or other`,
+          `commands creating output in your shell start script.`, ``,
+          `The connection to ${connection.currentConnectionName} will be aborted.`
+        ].join(`\n`),
+        modal: true
+      }, `Read more`);
+
+      if (chosen === `Read more`) {
+        commands.executeCommand(`open`, `https://codefori.github.io/docs/#/pages/tips/setup`);
```

Comment:
Link is not right. Updated it to also point to the relevant section

```suggestion
        commands.executeCommand(`open`, `https://codefori.github.io/docs/tips/setup/#error-in-shell-configuration`);
```